### PR TITLE
Python3 for 0.8.x

### PR DESCRIPTION
**Pull request by to266**
Similar to https://github.com/hyperspy/hyperspy/pull/873, but did not remove old-type slicing

### CODE REVIEW

File: hyperspy/drawing/utils.py [hunk 1]
```diff
@@ -956,7 +956,7 @@ def plot_spectra(
     if color is not None:
         if hasattr(color, "__iter__"):
             color = itertools.cycle(color)
-        elif isinstance(color, basestring):
+        elif isinstance(color, str):
```
Comment (vidartf) [inline hyperspy/drawing/utils.py]: This will never be reached since `str` has `__iter__`.

**Comment by vidartf** [comment]
Similar to #769 :)

**Comment by to266** [comment]
Fixed, thanks for the reminder and pointers :)

**Comment by vidartf** [comment]
Is the plan to merge this at/after release step, or to create a different branch?

**Comment by francisco-dlp** [comment]
I think that this shouldn't be merged. Once reviewed, this should be pushed to a new branch in the main repository e.g. 0.8.x_py3?

**Comment by vidartf** [comment]
@to266 : I just ran a `git fetch --all`, and it seems that you got all of my (now deleted) tags for trialing the appveyor deployment. I'm guessing it was when you cherry-picked commits in this branch? Not sure, but you might want to clean up your tags :)

**Comment by to266** [comment]
Thanks, removed those... And probably not all of them any way - I'm currently using at least one of them for exactly the reason :)

**Comment by to266** [comment]
If appveyor manages to build for py3.5, ready for review / merging / branching / whatever will happen...

in particular, @vidartf, could you have a look that the `appveyor.yml` is fine? thanks

**Comment by vidartf** [comment]
Well, you should probably merge latest version of 0.8.x for appveyor.yml to work with latest changes from #875.

**Comment by vidartf** [comment]
Looks OK to me, but I find it tricky to reliably judge such code by diff only. Why did you remove `natsort` from installation into deployment? My impression was that natsort was not part of the default WinPython distro?

File: appveyor.yml [hunk 2]
```diff
@@ -21,7 +40,7 @@ environment:
       CONDA_NPY: "19"
       WP_URL: 'http://heanet.dl.sourceforge.net/project/winpython/WinPython_2.7/2.7.10.3/WinPython-32bit-2.7.10.3.exe'
       WP_CRC: '99E825BCDE4FFA57E612A0D31DF1487861C96D1903E4D20C5843FA2651C9E33B'
-      DEPS: "numpy scipy matplotlib traits traitsui ipython h5py sympy scikit-learn dill mock"
+      DEPS: "numpy scipy matplotlib traits traitsui ipython h5py sympy scikit-learn dill mock natsort"
```
Comment (vidartf) [inline appveyor.yml]: Why is `natsort` added to the 2.7 dependencies?

Comment (vidartf) [inline appveyor.yml]: TBH, you can probably just delete the 2.7 entries. I set this up before there was a decision to not have a code base that works on both versions of python.

**Comment by to266** [comment]
That's exactly why I asked for your advice, thanks. Should be fixed.

**Comment by vidartf** [comment]
Have you done any test deployments on your branch? Probably cleaner if we don't muck about too much with test tags in the main repo.

**Comment by to266** [comment]
Nope, not yet.

I'll do it now

**Comment by to266** [comment]
FYI 
https://ci.appveyor.com/project/to266/hyperspy/build/1.0.24

**Comment by vidartf** [comment]
@to266 Not sure if that is an issue with pip, the WinPython included compiler, or traits. Maybe try updating the WinPython URLs (and checksums) to latest version. That should include a newer version of pip as well as a script to update pip correctly (patched, see https://github.com/winpython/winpython/commit/0b88f5c53e5306255fd3d9497480df2ef0665253). If the issue is with the compiler, we can switch to use the MSVC compilers that are installed on the AppVeyor servers, but I would prefer to use the WinPython ones for consistency if at all possible.

**Comment by to266** [comment]
Huh, still the same problem with the new winpython release (`pip` out of date):
https://ci.appveyor.com/project/to266/hyperspy/build/1.0.27/job/7fnk21yy7vv1oxjg

Also it looks like it fails to build `traits` wheel, not sure how relevant it is, as it stops at the pip step...

**Comment by to266** [comment]
Should we include this (https://github.com/winpython/winpython/commit/0b88f5c53e5306255fd3d9497480df2ef0665253) step?

**Comment by francisco-dlp** [comment]
Are you should that it is failing because there is a newer pip? May it not be failing to compile traits?

**Comment by to266** [comment]
no, I'm not. However that is the last command before crash. Any suggestions?

**Comment by francisco-dlp** [comment]
What about getting traits from http://www.lfd.uci.edu/~gohlke/pythonlibs/ instead to see if that works?

**Comment by vidartf** [comment]
Gohlke's page is designed to prevent automated/scripted fetching of files (cf. previous discussion about mayavi). If we're failing to build for 3.5 only, it might be an issue with traits, I just wanted to make sure that we use the latest compile tools from WinPython.

**Comment by vidartf** [comment]
PS: Which versions of Python 3 do we want to supply bundles for? Currently, the appveyor file was set up to test on 3.4 (64 bit) and 3.5 (32 bit). For deployment, I assume we want a 32 bit and 64 bit version for each version of Python 3. These extra versions should probably be set up to only run when there is a tag (i.e. to not have to many redundant test runs).

**Comment by to266** [comment]
Actually you mixed it up - 3.4 for 32 bit, and 3.5 for 64 bit, as winpython did not have 3.5 32 bit distribution before this (last) release - hence I changed them both to be 3.5. Any reason why not to go for 3.5 only? I'm not really experienced with 3.x pythons and probably would not be able to tell the difference anyway...

**Comment by vidartf** [comment]
Well, the first test is still running on Python 3.4 since it has `PYTHON: "C:\\Miniconda3"`, which is currently 3.4.

**Comment by vidartf** [comment]
BTW: I assume the correct path would be `PYTHON: "C:\\Miniconda35"`.

**Comment by vidartf** [comment]
> Any reason why not to go for 3.5 only?

For bundle, 3.5 only is OK. We should probably test on 3.4 as well, but it should probably be ok to add this in Travis only, as that runs multiple tests in parallel.

**Comment by to266** [comment]
Strange, it should be 3.5 for all:
http://conda.pydata.org/miniconda.html

**Comment by vidartf** [comment]
This is how AppVeyor has set up their images.

**Comment by francisco-dlp** [comment]
Shouldn't natsort be added to `setup.py`?

**Comment by to266** [comment]
@vidartf you're right: http://www.appveyor.com/docs/installed-software#python

I'll add/fix these things now, and have a look how to fix the build later (or just add a new tag just for the fun of it..)

**Comment by to266** [comment]
actually I just noticed one of them run fine:
https://github.com/to266/hyperspy/releases/tag/untagged-80aad89832b096f3fcae

Anyone keen to test it?

**Comment by vidartf** [comment]
> https://github.com/to266/hyperspy/releases/tag/untagged-80aad89832b096f3fcae

404?

**Comment by to266** [comment]
Deleted the tag, it was this:
https://ci.appveyor.com/project/to266/hyperspy/build/1.0.27/job/4wybqx797ti0y0lx/artifacts

**Comment by vidartf** [comment]
Seems like changing the dir fixed the issue. The question is why? Is there a cross-contamination between the test and deployment environments? Maybe we should inspect the `pip` command used in the deployment, and maybe call it by absolute value if need be? There might be an issue with `CMD_IN_ENV` ?

**Comment by to266** [comment]
In the commit before that I also added requirements for `traits` and `traitsui` (so that the versions are >= the first that supports python). Also before changing the directory `CONDA_PY` was already "35" - may have created the clash there, and now that everything is consistent, it worked fine and as expected

**Comment by vidartf** [comment]
PS: I see you switched to using the SHA1 checksum, not entirely sure why, but it means we lose some inherent protection from SHA256:

> File integrity checking is often intended to ensure that a file was downloaded correctly. But, if it is being used to verify that the file was not maliciously tampered with, you should consider an algorithm that is more resilient to collisions [than SHA1]. See also https://www.win.tue.nl/hashclash/SoftIntCodeSign/

**Comment by to266** [comment]
Because it was directly supplied by the winpython.
https://github.com/winpython/winpython/issues/264

I was also too lazy to download and hash them myself, but clearly it's not as important...

**Comment by to266** [comment]
There, fixed the laziness and now use `SHA256` (and it will hopefully be included with the next winpython release)

**Comment by to266** [comment]
https://github.com/hyperspy/hyperspy/tree/0.8.x_py3

**Comment by vidartf** [comment]
If pip ends up trying to fetch 0.8.4 on python 2, you could try adding a
python 2 wheel to 0.8.4. I think the wheel should be preferred? Or maybe
that was only universal wheels.

**Pull request merged**